Add merge request in GitLab plugin (#609)
Enalean/tuleap-documentation-en: languages/en/user-guide/code-versioning/gitlab.rst

    * both Git and GitLab plugins are installed and activated.
    * you have admin privileges in the Git service of your project.
    * you have a GitLab API token authorized to be used to query the GitLab API (see :ref:`gitlab-api-token`)

.. _gitlab-api-token:

GitLab API Token
````````````````

You can use a personal or project API token. The token will be used to manage integration of GitLab repository in Tuleap,
and to write comments automatically on GitLab commit or merge requests.

> 

Reviewer: Please add a note saying that the token must have the scope `api`

   :alt: GitLab repository tile
   :name: GitLab repository tile

From now on, each time you reference an artifact in a commit or merge request, a cross-reference will be created in the target artifact.

.. note::
  During the registration, a webhook is created in the GitLab repository.
  If the parameters of this webhook change (URL, events, or anything else), we cannot ensure that cross-references will
  continue to be created. See :ref:`gitlab-regenerate-webhook` to have more details.

Possible actions on GitLab repository
'''''''''''''''''''''''''''''''''''''

> As a Git administrator, go to the Git service of your project and find the repository that you want apply action.

Reviewer: ```suggestion
As a Git administrator, go to the Git service of your project and find the repository that you want to apply action.
```

When you click on cog icon in GitLab tile, you can:

* Edit access token
* Regenerate the GitLab webhook
* Unlink the repository

.. figure:: ../../images/screenshots/gitlab/tile-gitlab-dropdown.png
   :align: center
   :alt: Others actions on GitLab repository tile
   :name: Others actions on GitLab repository tile

Edit access token
`````````````````

If the token used during the integration has been revoked, you can change it by clicking on [Edit access token].
See :ref:`gitlab-api-token` to have more details.

.. figure:: ../../images/screenshots/gitlab/gitlab-edit-token-modal.png
   :align: center
   :alt: Editing GitLab API token
   :name: Editing GitLab API token

Confirm the action.

.. figure:: ../../images/screenshots/gitlab/gitlab-edit-token-modal-confirm.png
   :align: center
   :alt: Confirm editing GitLab API token
   :name: Confirm editing GitLab API token

.. note::

>   When you change API token, the :ref:`webhook is regenerated <gitlab-regenerate-webhook>` on GitLab side to be sure that the token is good.

Reviewer: No need to explain the technical details.
```suggestion
  When you change API token, the :ref:`webhook is also regenerated <gitlab-regenerate-webhook>` on GitLab side.
```

.. _gitlab-regenerate-webhook:

Regenerate the GitLab webhook
`````````````````````````````

> A webhook allows to communicate between GitLab server and Tuleap. This webhook is composed of a secret generated automatically by Tuleap

Reviewer: ```suggestion
A webhook allows GitLab to communicate with Tuleap. This webhook is composed of a secret generated by Tuleap
```

and some events (push and merge requests events).
If the webhook has been changed and is not functional, you can regenerate it by clicking on [Regenerate GitLab webhook].

.. figure:: ../../images/screenshots/gitlab/gitlab-regenerate-webhook-modal.png
   :align: center
   :alt: Modal to regenerate webhook
   :name: Modal to regenerate webhook

When the webhook is regenerated, the old is deleted from GitLab server, and a new webhook with a new secret is created.

Unregister repositories
```````````````````````

> If you want unregister a repository, you need to select [Unlink the repository] in the list. Then a modal opens and

Reviewer: ```suggestion
If you want to unregister a repository, you need to select [Unlink the repository] in the list. Then a modal opens and
```
